src/Orchard.Web/Modules/Orchard.Workflows/Controllers/AdminController.cs

```diff
@@ -303,6 +303,19 @@ IFormManager formManager
                 });
             }
 
+            foreach (var workflowRecord in workflowDefinitionRecord.WorkflowRecords) {

```

Comment:
@j3ffb I agree, it would be better

Moreover, if I well understood, here it seems that you can keep running instances even if an activity has been deleted in the definition. I did not analyze possible particular cases, but it would be nice. Maybe too complex to do the same even if an activity has been added :). Anyway, your implementation seems to be better

Best